docs/docs/architecture/security.md

```diff
@@ -0,0 +1,111 @@
+With aTLS, the party to be authenticated binds its TLS certificate to an attestation statement.
+For example, it embeds the certificate's public key into an attestation statement.
+Instead of relying on a certificate authority, aTLS uses this attestation statement to establish trust in the certificate.
+The protocol can be used by clients to verify a server certificate, by a server to verify a client certificate, or for mutual verification (mutual aTLS).
+
+
+## Encryption of state
+
+The Coordinator holds MarbleRun's state, which consists of the [manifest](../features/manifest.md), the [managed secrets](../features/secrets-management.md), and the [certificates for its CA](../features/attestation.md).
+The state is stored encrypted in persistent storage. For this, MarbleRun uses [AES128-GCM](https://www.rfc-editor.org/rfc/rfc5116#section-5.1) and a generated 16-byte Data Encryption Key (KEK) to seal the state to persistent storage.
+The DEK is also sealed to persistent storage to recover the state in case of a restart autonomously.
+[SGX sealing](https://www.intel.com/content/www/us/en/developer/articles/technical/introduction-to-intel-sgx-sealing.html) is used for that purpose.
+The Coordinator encrypts the DEK with a key encryption key (KEK).
+MarbleRun uses the SGX sealing key called `Product key` as its KEK, which is bound to its `Product ID` and the enclave's author `MRSIGNER` identity.
+In other words, a fresh and benign enclave instance of the same identity can recover that key.
+Hence, if the Coordinator is restarted on the same CPU, it can obtain the same KEK from the CPU, decrypt the DEK, and recover its state.
+
+If the Coordinator is restarted on a different CPU, it won't be able to obtain the same SGX sealing key from the CPU.
+Therefore, MarbleRun provides a [recovery feature](../features/recovery.md#recovery).
+The manifest allows for specifying a designated Recovery Key. The Recovery Key is a public RSA key. Upon startup, the Coordinator encrypts the DEK with this public key and returns it to the user.
+In case of a recovery event, the user decrypts the DEK locally and [uploads it to the Coordinator](../workflows/recover-coordinator.md).
+The Coordinator will decrypt the state with the DEK and proceed with operations.
+
+For [multi-party use cases](../features/recovery.md#multi-party-recovery), MarbleRun allows splitting the Recovery Key between parties.
+Every recovery party is defined in the manifest with its own public RSA key.
+The Coordinator generates a share of the recovery secret for every party and encrypts it with the corresponding RSA key.
+During a recovery event, every party will upload their share of the secret, which are all XORed together by the Coordinator to receive the combined key for decrypting the DEK.
+
+![Encrypted state single instance](../_media/enc-state-single.svg)
```

Comment:
The graphic should either be moved two paragraphs above or also illustrate recovery.